components/layout_2020/flexbox/layout.rs

```diff
@@ -301,19 +309,65 @@ impl FlexContainer {
         let line_interval = match flex_context.container_definite_inner_size.cross {
             Some(cross_size) if line_count >= 2 => {
                 let free_space = cross_size - content_cross_size;
+                let layout_is_flex_reversed = flex_context.flex_wrap_reverse;
 
                 cross_start_position_cursor = match flex_context.align_content {
+                    AlignContent::Start => Length::zero(),
+                    AlignContent::FlexStart => {
+                        if layout_is_flex_reversed {
+                            free_space
+                        } else {
+                            Length::zero()
+                        }
+                    },
+                    AlignContent::End => free_space,
+                    AlignContent::FlexEnd => {
+                        if layout_is_flex_reversed {
+                            Length::zero()
+                        } else {
+                            free_space
+                        }
+                    },
                     AlignContent::Center => free_space / 2.0,
-                    AlignContent::SpaceAround => free_space / (line_count * 2) as CSSFloat,
-                    AlignContent::FlexEnd => free_space,
-                    _ => Length::zero(),
+                    AlignContent::Stretch => Length::zero(),
+                    AlignContent::SpaceBetween => Length::zero(),
+                    AlignContent::SpaceAround => {
+                        if free_space >= Length::zero() {
+                            (free_space / line_count as CSSFloat) / 2.0
+                        } else {
+                            free_space / 2.0
```

Comment:
Ah, I was aware of the safe alignment `keyword`, but last I had checked the default change hadn't been widely implemented (or checked for web compatibility). Am I right in thinking that "falling back to safe centering" is equivalent to just falling back to `start` (as if we're falling back at all, then `safe center` will itself fall back to `start`)?

If we're going to change the default to safe alignment then I'm thinking we should probably also implement the keywords themselves. Which will likely mean that we want to switch over to Gecko's representation of the alignment styles in Stylo (and we may also want to switch the `-items` and `-self` properties).

My preference would be to land this first (it being an improvement on what we had) and make the larger change as a follow up PR. But if you'd prefer to jump straight to that implementation in one big leap then I could also do that. What do you think?

Reply:
> Am I right in thinking that "falling back to safe centering" is equivalent to just falling back to start

I think that's the case. Well, maybe if the flex container is a scroll container, the start scroll position might be different (https://drafts.csswg.org/css-align/#overflow-scroll-position) but I think nobody implements this well.

> But if you'd prefer to jump straight to that implementation in one big leap then I could also do that

No, I didn't mean to make the larger changes here, I just meant using `Length::zero()` instead of `free_space / 2.0`.